docs/advanced-topics/networkobject-parenting.md

```diff
@@ -0,0 +1,173 @@
+
+### Only Reparenting During Networking Is Valid
+
+A `NetworkObject` can only be reparented while networking, in other terms you can only reparent while listening/running as a server.
+
+If you allowed moves while not networking, you would be desynced immediately upon switching to networking. Also reparenting a `NetworkObject` under a non-`NetworkObject` while not networking would sound valid but that would not be replicable on the remote-side since Netcode does not cover full scene hierarchy synchronization.
+
+
+### Invalid Reparenting Will Move `NetworkObject` Back To Its Original Location
+
+If an invalid/unsupported `NetworkObject` parenting happens, Netcode will immediately pop it back to its previous location to keep things in sync and also will provide relevant error/warning messages to indicate the issue.
+
+## (Re)Parenting Move Examples
+
+We will assume that our initial scene hierarchy is looking like this:
+
+```
+Sun
+Tree
+Camera
+Player (NetworkObject)
+  ├─ Head
+  ├─ Body
+  ├─ Arms
+  │  ├─ LeftArm
+  │  │  └─ LeftHand (NetworkObject)
+  │  └─ RightArm
+  │     └─ RightHand (NetworkObject)
+  └─ Legs
+Axe (NetworkObject)
+```
+
+So, let's try a few moves!
+
+### Root/Axe → RightHand/Axe
+
+This is a **valid** move because `Axe (NetworkObject)` is being moved under `RightHand (NetworkObject)`. We know about their `NetworkObjectId`s and it will be replicated across the network to the clients by the server.
+
+Now, our hierarchy is looking like this:
+
+```
+Sun
+Tree
+Camera
+Player (NetworkObject)
+  ├─ Head
+  ├─ Body
+  ├─ Arms
+  │  ├─ LeftArm
+  │  │  └─ LeftHand (NetworkObject)
+  │  └─ RightArm
+  │     └─ RightHand (NetworkObject)
+  │        └─ Axe (NetworkObject) [to] <──┐
+  └─ Legs                                 ├ OK
+                                [from] ───┘
+```
+
+### RightHand/Axe → Body/Axe
+
+This is an **invalid** move because `Axe (NetworkObject)` is being moved under `Body` which is _not_ a `NetworkObject`. It does _not_ have a `NetworkObjectId` and it can _not_ be replicated and synced on the clients.
+
+So, we tried to do this but it did _not_ succeed:
+
+```
+Sun
+Tree
+Camera
+Player (NetworkObject)
+  ├─ Head
+  ├─ Body
+  │                               [to] <──┐
+  ├─ Arms                                 │
+  │  ├─ LeftArm                           │
+  │  │  └─ LeftHand (NetworkObject)       ├ INVALID
+  │  └─ RightArm                          │
+  │     └─ RightHand (NetworkObject)      │
+  │        └─ Axe (NetworkObject) [from] ─┘
+  └─ Legs
+```
+
+We'd get an error message in the logs similar to this:
+
+```
+Invalid parenting, NetworkObject moved under a non-NetworkObject parent
+```
+
+### RightHand/Axe → SceneRoot/Axe
+
+This is a **valid** move because `Axe (NetworkObject)` is being moved to the scene root (no parent). Even though there is no `NetworkObjectId` to sync, empty/null parent _can_ be synced across the network on the clients.
```

Comment:
```suggestion
This is a **valid** move because `Axe (NetworkObject)` is being moved to the scene root (no parent). So even though there is no `NetworkObjectId` to sync, empty/null parent _can_ be synced across the network on the clients.
```